### ISSUE TITLE
CTRL+D at shell prompt causes endless flurry of "child terminated"

### ISSUE THREAD

**Issue by egranata** [egranata/puppy#43]
To reproduce, press CTRL+D while at the shell prompt and observe processes spawn and die without any end in sight..

**Comment by egranata** [comment]
This works up until PID 999 and then it causes a GPF and subsequent kernel panic

Turns out, operator error. Everywhere in the OS assumes there are 2000 different process IDs one can have in flight in parallel, but the truth is:
```
    ; task segments [6...]
	times 2000 dd 0
```
there are only 1000 such selectors (GDT selectors are 8 bytes not 4)

And in fact, the fault code is 0x1F70, which - when shifted as per https://wiki.osdev.org/Exceptions#Selector_Error_Code - becomes selector 1006 (selectors 0 thru 5 are reserved for kernel functions, so 6 thru 1005 are usable process selectors and 1006 is - well - not good)

Couple of things one could do:
- well, first, either have 1000 processes or have 2000 selectors;
- it is nice to have incrementing PIDs, but GDT selectors are an implementation detail of the task switcher, so in theory one could keep recycling them (although doing this would have masked the issue until one truly had 1000 parallel processes)
